pkg/kubelet/kubelet.go

```diff
@@ -2770,16 +2770,19 @@ func (kl *Kubelet) setNodeStatusMachineInfo(node *api.Node) {
 }
 
 // Set versioninfo for the node.
-func (kl *Kubelet) setNodeStatusVersionInfo(node *api.Node) {
+func (kl *Kubelet) setNodeStatusVersionInfo(node *api.Node) error {
 	verinfo, err := kl.cadvisor.VersionInfo()
 	if err != nil {
 		glog.Errorf("Error getting version info: %v", err)
+		return err
 	} else {
 		node.Status.NodeInfo.KernelVersion = verinfo.KernelVersion
 		node.Status.NodeInfo.OSImage = verinfo.ContainerOsVersion
 
 		runtimeVersion := "Unknown"
-		if runtimeVer, err := kl.containerRuntime.Version(); err == nil {
+		if runtimeVer, err := kl.containerRuntime.Version(); err != nil {
+			return err
+		} else {
```

Comment:
I have pushed an updated version but I feel that "if-else" would have been better as it does not increase the scope of runtimeVer and err beyond if-else as that is not needed.